security/acme-client/src/opnsense/mvc/app/views/OPNsense/AcmeClient/actions.volt

```diff
@@ -60,7 +60,135 @@ POSSIBILITY OF SUCH DAMAGE.
                 $(".method_table_"+$(this).val()).show();
             });
             $("#action\\.type").change();
-        })
+        });
+
+        // Helpers for extra buttons and status divs
+        function makeButton(label, buttonGroup, buttonClass) {
+            var button = $('<button class="btn" type="button">'
+                           + '<span class="btn-text"></span>'
+                           + '<i class="fa fa-spinner fa-pulse" style="margin-left: 0.5em;"></i></button>');
+            button.addClass(buttonClass || "btn-primary");
+            $('.fa-spinner', button).hide();
+            $('.btn-text', button).html(label);
+
+            var targetContainer = $("#DialogAction .modal-footer"),
+                targetId = "method_table_" + buttonGroup,
+                target = $("." + targetId, targetContainer);
+
+            if (!target.is('span')) {
+                target = $('<span class="method_table" style="float: left"></span>')
+                    .addClass(targetId)
+                    .prependTo(targetContainer)
+                    .hide();
+            }
+
+            return button.appendTo(target);
+        }
+
+        function makeStatusDiv(anchor, statusClass) {
+            return $('<div class="alert method_table" role="alert" style="word-break: break-all"></div>')
+                .appendTo($(anchor).closest("table").find("thead th[colspan=3]").first())
+                .addClass(statusClass || 'alert-info')
+                .hide();
+        }
+
+        // SFTP - Identity show button
+        (function ($identityType) {
+            var identityDiv = makeStatusDiv($identityType);
+
+            makeButton("{{ lang._('Show Identity') }}", "upload_sftp", "btn-info")
+                .click(function () {
+                    identityDiv.hide();
+                    var button = $(this);
+                    button.prop('disabled', true).find(".fa-spinner").show();
+
+                    ajaxCall("/api/acmeclient/actions/sftpGetIdentity", getFormData("DialogAction").action, function (data, status) {
+                        button.prop('disabled', false).find(".fa-spinner").hide();
+
+                        if (status === "success" && data.status === "ok")
+                            identityDiv.text(data.identity).show();
+                        else
+                            identityDiv.text("{{ lang._('Failed loading identity') }}").show();
```

Comment:
to avoid problems I suggest adding curly brackets

Reply:
ok